src/compat/_internal/memoizeCapped.ts

```diff
@@ -0,0 +1,15 @@
+import { memoize } from '../../function';
+
+export function memoizeCapped<F extends (...args: Parameters<F>) => unknown>(func: F) {
```

Comment:
Personally, I really didn't want to add this function to `es-toolkit/compat`.. Does this make a big difference when it comes to performance?

Reply:
<img width="627" alt="Screenshot 2024-08-25 at 10 52 09 PM" src="https://github.com/user-attachments/assets/fc8287c2-ceeb-4205-8f9c-d05784008e57">

This is result that not using it.

I think that it is good for performance.

Reply:
<img width="776" alt="Screenshot 2024-08-25 at 11 14 48 PM" src="https://github.com/user-attachments/assets/39858a99-4744-4956-bcb3-dbdf1385e476">

I removed it!

Now, we just decrease counts of calling `toPath` function instead of memoizing.

Reply:
While changing the logic, I realized that the existing code had too many conditional statements, so I also proceeded with refactoring.